src/os/bluestore/ZonedAllocator.cc

```diff
@@ -105,6 +105,13 @@ uint64_t ZonedAllocator::get_free() {
   return num_free;
 }
 
+bool ZonedAllocator::zoned_needs_cleaning() {
+  std::lock_guard l(lock);
+  // Trigger cleaning if less than 25% is free.
+  // TODO: Make this a tunable.
+  return static_cast<double>(num_free) / size < 0.25;
```

Comment:
By having num_free atomic one might avoid tacking lock in this method

Reply:
Or perhaps even better to merge zoned_needs_cleaning() and zoned_get_zones_to_clean() logic into a single method which both checks for the need to clean zones and return the zone list if needed. The "checking" stage can be 
performed on an atomic (without a lock acquisition) anyway...

Reply:
Turned num_free to atomic and merged the two functions into one.